app/controllers/messages_controller.rb

```diff
@@ -6,10 +6,6 @@ class MessagesController < ApplicationController
     controller.ensure_logged_in t("layouts.notifications.you_must_log_in_to_send_a_message")
   end
 
-  before_filter do |controller|
-    controller.ensure_authorized t("layouts.notifications.you_are_not_authorized_to_do_this")
-  end
-
```

Comment:
The main problem here is that the URL schema is a bit wrong. For example, in my opinion, instead of /username/settings/, the settings page should live in /settings. That's for example how Facebook does it: www.facebook.com/username is the URL for the profile page (same as we do) but then www.facebook.com/settings is where you can access settings (not www.facebook.com/username/settings). So in the future, I'd like to change the URL schema to that direction. 

So what we currently do with the username in the URL? We don't care. We update the settings of the current_user, even if I send a POST request to /ovan/settings